src/main/java/org/springframework/data/elasticsearch/core/SearchOperations.java

```diff
@@ -170,32 +202,203 @@ default long count(Query query, IndexCoordinates index) {
+			Class<T> clazz);
 
 	/**
 	 * Clears the search contexts associated with specified scroll ids.
 	 *
 	 * @param scrollId the scroll id
+	 * @deprecated since 4.0, use {@link #searchScrollClear(String)}.
 	 */
-	void clearScroll(String scrollId);
+	@Deprecated
+	default void clearScroll(String scrollId) {
+		searchScrollClear(scrollId);
+	}
 
 	/**
 	 * more like this query to search for documents that are "like" a specific document.
 	 *
+	 * @param <T> element return type
+	 * @param query the query to execute
+	 * @param clazz the entity clazz used for property mapping
+	 * @param index the index to run the query against
+	 * @return page with the results
+	 * @deprecated since 4.0, use {@link #searchMoreLikeThis(MoreLikeThisQuery, Class, IndexCoordinates)}.
+	 */
+	@Deprecated
+	default <T> AggregatedPage<T> moreLikeThis(MoreLikeThisQuery query, Class<T> clazz, IndexCoordinates index) {
+		return (AggregatedPage<T>) SearchHitSupport.unwrapSearchHits(searchMoreLikeThis(query, clazz, index));
+	}
+
+	/**
+	 * Execute the query against elasticsearch and return the first returned object.
+	 *
+	 * @param query the query to execute
+	 * @param clazz the entity clazz used for property mapping
+	 * @param index the index to run the query against
+	 * @return the first matching object
+	 */
+	default <T> SearchHit<T> searchForObject(Query query, Class<T> clazz, IndexCoordinates index) {
+		List<SearchHit<T>> content = searchForPage(query, clazz, index).getContent();
+		return content.isEmpty() ? null : content.get(0);
+	}
+
+	/**
+	 * Execute the query against elasticsearch and return result as {@link Page}.
+	 *
+	 * @param query the query to execute
+	 * @param clazz the entity clazz used for property mapping
+	 * @param index the index to run the query against
+	 * @return a page with aggregations
+	 */
+	<T> AggregatedPage<SearchHit<T>> searchForPage(Query query, Class<T> clazz, IndexCoordinates index);
+
+	/**
+	 * Execute the multi-search against elasticsearch and return result as {@link List} of {@link Page}
+	 *
+	 * @param queries the queries
+	 * @param clazz the entity clazz used for property mapping
+	 * @param index the index to run the query against
+	 * @return list of pages with the results
+	 */
+	<T> List<Page<SearchHit<T>>> searchForPage(List<? extends Query> queries, Class<T> clazz, IndexCoordinates index);
+
+	/**
+	 * Execute the multi-search against elasticsearch and return result as {@link List} of {@link Page}
+	 *
+	 * @param queries the queries
+	 * @param classes the entity classes used for the queries
+	 * @param index the index to run the query against
+	 * @return list of pages with the results
+	 */
+	List<AggregatedPage<? extends SearchHit<?>>> searchForPage(List<? extends Query> queries, List<Class<?>> classes,
+			IndexCoordinates index);
+
+	/**
+	 * Execute the criteria query against elasticsearch and return result as {@link List}
```

Comment:
We should also cleanup the Javadoc before cleanup (`SearchHits` instead of list). Check also the other methods.